src/services/wallet.service.ts

```diff
@@ -71,11 +75,23 @@ export class WalletService {
     })
 
     window.ethereum.on('chainChanged', (chainId) => {
+      if (this.networkId === chainId) {
+        return
+      }
       if (CHAIN_ID_LIST[Number(chainId)]) {
-        store.dispatch({
-          type: 'SETUP/CHAINIDCHANGED/SET',
-          payload: Number(chainId),
-        })
+        if (!this.userTriggeredSwitchChain) {
+          store.dispatch({
+            type: 'SETUP/CHAINIDCHANGED/SET',
+            payload: Number(chainId),
+          })
+        } else {
+          store.dispatch({
+            type: 'SETUP/USER_TRIGGERED_CHAIN_SWITCH/SET',
+            payload: true,
+          })
+        }
```

Comment:
Can you please reverse the condition so we don't have to think it loud and can grab it easily.

```suggestion
        if (this.userTriggeredSwitchChain) { // just check for true directly.
          store.dispatch({
            type: 'SETUP/USER_TRIGGERED_CHAIN_SWITCH/SET',
            payload: true,
          })
        } else {
          store.dispatch({
            type: 'SETUP/CHAINIDCHANGED/SET',
            payload: Number(chainId),
          })
        }
```